code/game/objects/structures/watercloset.dm

```diff
@@ -227,8 +228,16 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/urinal, 32)
 		exposed = !exposed
 	return TRUE
 
+/obj/structure/urinal/wrench_act_secondary(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(user)
+	deconstruct(TRUE)
+	to_chat(user, span_notice("You remove [src] from the wall."))
+	return TRUE
+
 /obj/structure/urinal/deconstruct(disassembled = TRUE)
 	if(!(obj_flags & NO_DECONSTRUCTION))
+		for(var/obj/urinal_item in contents)
+			urinal_item.forceMove(drop_location())
```

Comment:
This should just be `hidden_item?.forceMove(drop_location())`, no?

Also: You need to implement `/obj/structure/urinal/Exited(...)` to properly clear refs on hidden item

Reply:
I don't know how hidden_item? works, I tested it to see if it worked and yes it did, just to realize that I forgot the question mark, I added it and it works the same, but does it actually matter or not?
As for the second implementation, I have no idea what refs are but I'll try to add it in? Could you help explain it to me a little bit?
Oh yeah small explanation I am really new and have basically just about no idea what I'm doing.

Reply:
> I don't know how hidden_item? works, I tested it to see if it worked and yes it did, just to realize that I forgot the question mark, I added it and it works the same, but does it actually matter or not?

the `?` checks that the item exists before trying to call stuff on it: https://www.byond.com/docs/ref/#/operator/%3f%2e

You would need it here because otherwise the proc would runtime if one is deconstructed without a hidden item inside